applications/mlflow/chart/mlflow/templates/ingress.yaml

```diff
@@ -0,0 +1,69 @@
+{{- if .Values.mlflow.ingress.enabled -}}
+{{- if and .Values.mlflow.ingress.className (not (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion)) }}
+  {{- if not (hasKey .Values.ingress.annotations "kubernetes.io/ingress.class") }}
+  {{- $_ := set .Values.mlflow.ingress.annotations "kubernetes.io/ingress.class" .Values.mlflow.ingress.className}}
+  {{- end }}
+{{- end }}
+{{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1
+{{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1beta1
+{{- else -}}
+apiVersion: extensions/v1beta1
+{{- end }}
```

Comment:
I love having an example of this in here. I would tend to move it to a helper function so that you don't have to do this multiple times but maybe there's only one ingress here so this is fine.

Reply:
yeah, only one ingress at the moment. i'll make a note to evaluate what code across the entire chart we consolidate as a future chore sort of item.